edge-modules/edge-dashboard/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+    _req: HttpRequest,
+    device: web::Data<Option<state::Device>>,
+    // _info: web::Query<AuthRequest>,
+) -> HttpResponse {
+    // println!("Query string version: {}", info.version);
+
+    if let Some(dev) = device.get_ref() {
+        println!("Dev: {:?}", dev);
+        state::return_response(&dev)
+    } else {
+        HttpResponse::UnprocessableEntity().body("Device connection string unable to be processed.")
+    }
+}
+
+fn get_modules(
+    context: web::Data<Arc<Context>>,
+    info: web::Query<AuthRequest>,
+) -> Box<dyn Future<Item = HttpResponse, Error = ActixError>> {
+    let api_ver = &info.api_version;
+    return_modules(context, api_ver, module_response)
+}
+
+fn get_connectivity(_req: HttpRequest, device: web::Data<Option<state::Device>>) -> HttpResponse {
+    if let Some(dev) = device.get_ref() {
+        if let Some(iothub) = dev.hub_name() {
+            let iothub_hostname = &format!("{}.azure-devices.net", iothub);
+
+            let r = resolve_and_tls_handshake(&(&**iothub_hostname, 443), iothub_hostname);
+            match r {
+                Ok(_) => HttpResponse::Ok().body("Connected with IoT Hub!"),
+                Err(_) => HttpResponse::UnprocessableEntity()
+                    .body("Failed to establish connection with IoT Hub."),
+            }
+        } else {
+            HttpResponse::UnprocessableEntity().body("IoT Hub name could not be processed")
+        }
+    } else {
+        HttpResponse::UnprocessableEntity().body("IoT Hub name could not be processed")
+    }
+}
+
+fn get_health(
+    _req: HttpRequest,
+    context: web::Data<Arc<Context>>,
+    info: web::Query<AuthRequest>,
+) -> Box<dyn Future<Item = HttpResponse, Error = ActixError>> {
+    let api_ver = &info.api_version;
+    return_modules(context, api_ver, health_response)
+}
+
+fn restart_module(req: HttpRequest,
+    context: web::Data<Arc<Context>>,
+    info: web::Query<AuthRequest>
+) -> Box<dyn Future<Item=HttpResponse, Error=ActixError>> {
+    
+    let module_id = req.match_info().get("id");
+    let api_ver = &info.api_version;
+    let response = context
+        .edge_config
+        .as_ref()
+        .map(|config| {
+            let mgmt_uri = config.connect().management_uri();
+            // println!("API Version: {}", api_ver);
+            Either::A(
+                Url::parse(&format!("{}/modules/?api-version={}", mgmt_uri, api_ver))
+                    .map_err(ErrorInternalServerError)
+                    .and_then(|url| ModuleClient::new(&url).map_err(ErrorInternalServerError)) // can't connect to the endpoint
+                    .map(|mod_client| {
+                        if let Some(id) = module_id {
+                            mod_client
+                            .restart(id)
+                        } else {
+                            mod_client.restart("")
+                        }
+                        .map(|_| HttpResponse::Ok().body("Module was successfully restarted."))
```

Comment:
You don't need response data in an OK Response unless it's specific data related to the request.